lib/client/redirect.go

```diff
@@ -145,15 +145,15 @@ func (rd *Redirector) Start() error {
 	query.Set("secret_key", rd.key.String())
 	u.RawQuery = query.Encode()
 
-	out, err := rd.proxyClient.PostJSON(rd.context, rd.proxyClient.Endpoint("webapi", rd.Protocol, "login", "console"), SSOLoginConsoleReq{
+	out, err := rd.proxyClient.PostJSONWithFallback(rd.context, rd.proxyClient.Endpoint("webapi", rd.Protocol, "login", "console"), SSOLoginConsoleReq{
```

Comment:
This `PostJSON` call posts directly to the Teleport proxy (e.g., `https://teleport-root.sqprod.co:443/`). This endpoint actually resolves to our authenticating BeyondCorp proxy, through which direct HTTP calls are made, and it forwards them to the correct backend.

However, it expects connections through it to be made over HTTP. The connection *to* the proxy is over TLS, but the HTTP requests within must be made over plain HTTP, rather than over TLS to the backend using an HTTP `CONNECT` to establish a raw TCP tunnel. As such, we need `PostJSON` to fall back to HTTP when `--insecure` is passed.